pyomo/contrib/doe/doe.py

```diff
@@ -1190,18 +1200,20 @@ def _fix_design(self, m, design_val, fix_opt=True, optimize_option=None):
             # Get Pyomo variable object
             cuid = pyo.ComponentUID(name)
             var = cuid.find_component_on(m)
-            if fix_opt:
-                # If fix_opt is True, fix the design variable
-                var.fix(design_val[name])
-            else:
-                # Otherwise check optimize_option
-                if optimize_option is None:
-                    # If optimize_option is None, unfix all design variables
-                    var.unfix()
+
+            if var.ctype is Var:
+                if fix_opt:
+                    var.fix(design_val[name])
                 else:
-                    # Otherwise, unfix only the design variables listed in optimize_option with value True
-                    if optimize_option[name]:
+                    if optimize_option is None:
                         var.unfix()
+                    else:
+                        if optimize_option[name]:
+                            var.unfix()
+
+            elif var.ctype is Param:
+                var = design_val[name]
```

Comment:
What if this is called with `fix_opt=False`? Are there cases where the user expects var to be a degree of freedom and they would be surprised by this code silently leaving it as fixed? If this is expected behavior then I think it needs to be clear in the documentation what the tradeoff is when using a design variable declared as a Pyomo Param vs a Pyomo Var.

Reply:
if with `fix_opt=False`, var will be unfixed, so it is fine if the user wants to leave it as DOF (they can use fix_opt=False to do that). I add a statement in the doc to note that, if a design variable is defined as `Param`, `stochastic_program` function will not work because it cannot be optimized.